runtime/gateway_test.go

```diff
@@ -99,15 +99,18 @@ func TestGetServiceNameFromEnv(t *testing.T) {
 		MetricsBackend:            metricsBackend,
 	}
 
-	os.Setenv("TEST", "overridden")
-
+	if err := os.Setenv("TEST", "overridden"); err != nil{
+		t.Errorf("no error expected but got %v", err)
+	}
 	g1, err := CreateGateway(cfg, opts)
 	assert.Nil(t, err)
 	assert.Equal(t, g1.ServiceName, "overridden")
-	os.Unsetenv("TEST")
+	if err = os.Unsetenv("TEST"); err != nil{
```

Comment:
Go linter reports warnings because `os.Unsetenv` returns an error which was not tested against. Hence adding these checks